apps/ledger-live-desktop/src/renderer/screens/settings/sections/Developer/OnboardingAppInstallDebug.tsx

```diff
@@ -0,0 +1,156 @@
+
+import ButtonV2 from "~/renderer/components/Button";
+import { setDrawer } from "~/renderer/drawers/Provider";
+import OnboardingAppInstallStep from "~/renderer/components/OnboardingAppInstall";
+import { withV3StyleProvider } from "~/renderer/styles/StyleProviderV3";
+import { getCurrentDevice } from "~/renderer/reducers/devices";
+
+const defaultRestoreDevice: DeviceModelInfo = {
+  modelId: DeviceModelId.nanoX,
+  deviceInfo: {} as DeviceInfo,
+  apps: [
+    { name: "Ethereum", version: "" },
+    { name: "Polygon", version: "" },
+  ],
+};
+
+type SelectRestoreDeviceItem = { value: DeviceModelId | null; label: string };
+
+const restoreDeviceOptions: SelectRestoreDeviceItem[] = [
+  { value: null, label: "Use default apps" },
+  { value: DeviceModelId.nanoX, label: "Restore from Nano X" },
+  { value: DeviceModelId.nanoSP, label: "Restore from Nano S Plus" },
+  { value: DeviceModelId.nanoS, label: "Restore from Nano S" },
+  { value: DeviceModelId.stax, label: "Restore from Ledger Stax" },
+];
+
+const OnboardingAppInstallDebugScreen = () => {
+  const [restore, setRestore] = useState<boolean>(false);
+  const [componentKey, setComponentKey] = useState<number>(1);
+  const [installDone, setInstallDone] = useState<boolean>(false);
+  const [restoreDevice, setRestoreDevice] = useState<DeviceModelInfo>(defaultRestoreDevice);
+  const [
+    selectedRestoreDeviceOption,
+    setSelectedRestoreDeviceOption,
+  ] = useState<SelectRestoreDeviceItem | null>(restoreDeviceOptions[0]);
+  const [appsToRestore, setAppsToRestore] = useState<string[]>(
+    defaultRestoreDevice.apps.map(app => app.name),
+  );
+  const device = useSelector(getCurrentDevice);
+
+  useEffect(() => {
+    setRestoreDevice(prev => ({
+      ...prev,
+      apps: appsToRestore.map(app => ({ name: app, version: "" })),
+    }));
+  }, [appsToRestore]);
+
+  useEffect(() => {
+    if (selectedRestoreDeviceOption) {
+      setRestore(selectedRestoreDeviceOption.value !== null);
+      if (selectedRestoreDeviceOption.value) {
+        setRestoreDevice(prev => ({
+          ...prev,
+          modelId: selectedRestoreDeviceOption.value as DeviceModelId,
+        }));
+      }
+    }
+  }, [selectedRestoreDeviceOption]);
+
+  const handleRemount = useCallback(() => {
+    setComponentKey(prev => prev + 1);
+    setInstallDone(false);
+    setSelectedRestoreDeviceOption(restoreDeviceOptions[0]);
+    setAppsToRestore(defaultRestoreDevice.apps.map(app => app.name));
+    setRestoreDevice(defaultRestoreDevice);
+  }, []);
+
+  const steps = [
+    {
+      key: 0,
+      status: "completed",
+      title: "Seed your device",
+    },
+    {
+      key: 1,
+      estimatedTime: 120,
+      status: installDone ? "completed" : "active",
+      title: "Install default set of apps",
+      renderBody: () => (
+        <OnboardingAppInstallStep
+          restoreDevice={restore ? restoreDevice : undefined}
+          device={device as Device}
```

Comment:
Why lie to TypeScript here? `OnboardingAppInstallStep` actually accepts `Device | null` for this prop.
We should avoid `as` as much as possible

Reply:
Done